tools/serverpod_cli/bin/generator/protocol_generator.dart

```diff
@@ -53,162 +58,127 @@ abstract class ProtocolGenerator {
+                            literalString(param.name): refer(
+                                    'ParameterDescription', serverPodUrl(true))
+                                .call([], {
+                              'name': literalString(param.name),
+                              'type': refer('getType', serverPodUrl(true))
+                                  .call([], {}, [param.type.reference(true)]),
+                              'nullable': literalBool(param.type.nullable),
+                            })
+                        }),
+                        'call': Method(
+                          (m) => m
+                            ..requiredParameters.addAll([
+                              Parameter((p) => p
+                                ..name = 'session'
+                                ..type = refer('Session', serverPodUrl(true))),
+                              Parameter((p) => p
+                                ..name = 'params'
+                                ..type = TypeReference((t) => t
+                                  ..symbol = 'Map'
+                                  ..types.addAll([
+                                    refer('String'),
+                                    refer('dynamic'),
+                                  ])))
+                            ])
+                            ..modifier = MethodModifier.async
+                            ..body = refer('endpoints')
+                                .index(literalString(endpoint.name))
+                                .asA(refer(
+                                    endpoint.className, endpointPath(endpoint)))
+                                .property(method.name)
+                                .call([
+                              refer('session'),
+                              for (var param in [
+                                ...method.parameters,
+                                ...method.parametersPositional
+                              ])
+                                refer('params')
+                                    .index(literalString(param.name)),
+                            ], {
+                              for (var param in [...method.parametersNamed])
+                                param.name: refer('params')
+                                    .index(literalString(param.name)),
+                            }).code,
+                        ).closure,
+                      }),
+                  })
+                }))
+                .statement,
+          // Hook up modules
+          for (var module in config.modules)
+            refer('modules')
+                .index(literalString(module.name))
+                .assign(refer('Endpoints',
+                        'package:${module.serverPackage}/module.dart')
+                    .call([])
+                    .cascade('initializeEndpoints')
+                    .call([refer('server')]))
+                .statement,
+        ])))));
```

Comment:
I would add trailing commas here for readability. If indentation gets too heavy, break it down into sub-methods.

Reply:
c9c5c35 Like this? Or sub methods?

Reply:
Yes, like this. :) I'd say it's a borderline case for readability. It may benefit from breaking it down into some sub-methods if possible.

Reply:
Fixed on new branch.